pkg/api/v20170701/validate.go

```diff
@@ -106,36 +89,59 @@ func (a *AgentPoolProfile) Validate(orchestratorType OrchestratorType) error {
 			return fmt.Errorf("AgentPoolProfile.Ports must be empty when AgentPoolProfile.DNSPrefix is empty")
 		}
 	}
-	if e := validateStorageProfile(a.StorageProfile); e != nil {
-		return e
-	}
 	return nil
 }
 
 // Validate implements APIObject
 func (l *LinuxProfile) Validate() error {
-	if e := validateName(l.AdminUsername, "LinuxProfile.AdminUsername"); e != nil {
-		return e
-	}
-	if len(l.SSH.PublicKeys) != 1 {
-		return errors.New("LinuxProfile.PublicKeys requires only 1 SSH Key")
-	}
-	if e := validateName(l.SSH.PublicKeys[0].KeyData, "LinuxProfile.PublicKeys.KeyData"); e != nil {
-		return e
+	// Don't need to call validate.Struct(l)
+	// It is handled by Properties.Validate()
+	if e := validate.Var(l.SSH.PublicKeys[0].KeyData, "required"); e != nil {
+		return fmt.Errorf("KeyData in LinuxProfile.SSH.PublicKeys cannot be empty string")
 	}
 	return nil
 }
 
+func handleValidationErrors(e validator.ValidationErrors) error {
```

Comment:
@JackQuincy @JiangtianLi @weinong 
Now, this function will be called to handle the `validator.ValidationErrors`, it is called only in one place.
I noticed, I have a few hard coded strings here. However, it will be no difference with before, when we call
```
validateName(l.AdminUsername, "LinuxProfile.AdminUsername")
```
We also need to pass a hard coded string as a parameter.

This function is better in the sense, it has all hard-code strings in one place.